test/com/google/javascript/jscomp/RewriteDynamicImportsTest.java

```diff
@@ -451,4 +456,162 @@ public void outputModulesInternalImportDifferentChunks_used() {
 
     test(srcs(new JSChunk[] {actualChunk0, actualChunk1}), expectedSrcs);
   }
+
+  @Test
+  public void aliasExternInjectedSimple() {
+    language = LanguageMode.ECMASCRIPT_2015;
+    languageIn = LanguageMode.ECMASCRIPT_NEXT;
+    this.dynamicImportAlias = "import_";
+    JSChunk actualChunk0 = new JSChunk("chunk0");
+    actualChunk0.add(SourceFile.fromCode("i0.js", "const a = 1; export default a;"));
+
+    JSChunk actualChunk1 = new JSChunk("chunk1");
+    actualChunk1.add(SourceFile.fromCode("i1.js", "const nsPromise = import('./i0.js');"));
+
+    testExternChanges(
+        "",
+        new JSChunk[] {actualChunk0, actualChunk1},
+        "function import_(specifier) {}",
+        (DiagnosticType[]) null);
+  }
+
+  @Test
+  public void aliasExternInjectedSimpleImport() {
+    language = LanguageMode.ECMASCRIPT_2015;
+    languageIn = LanguageMode.ECMASCRIPT_NEXT;
+    this.dynamicImportAlias = "import";
+
+    // "import" is not a valid JS identifier name and will not parse.
+    // Instead, we have to manually check the AST for the expected name.
+    Compiler compiler = createCompiler();
+    CompilerOptions options = getOptions();
+    compiler.init(
+        ImmutableList.of(SourceFile.fromCode(GENERATED_EXTERNS_NAME, "")),
+        ImmutableList.of(
+            SourceFile.fromCode(
+                GENERATED_SRC_NAME,
+                "var external = 'url'; const nsPromise = import(external);")),
+        options);
+    compiler.parseInputs();
+    new GatherModuleMetadata(
+        compiler, /* processCommonJsModules= */ false, ResolutionMode.BROWSER)
+        .process(compiler.getExternsRoot(), compiler.getJsRoot());
+    new ModuleMapCreator(compiler, compiler.getModuleMetadataMap())
+        .process(compiler.getExternsRoot(), compiler.getJsRoot());
+    assertThat(compiler.getErrors()).isEmpty();
+    Node externsAndJs = compiler.getRoot();
+    Node externs = externsAndJs.getFirstChild();
+    Node root = externsAndJs.getLastChild();
+    compiler.beforePass(this.getName());
+    getProcessor(compiler).process(externs, root);
+
+    Node syntheticExternsAtEnd = externs.getLastChild();
+    assertThat(syntheticExternsAtEnd).isNotNull();
+    assertThat(syntheticExternsAtEnd.isScript()).isTrue();
+    Node injectedAlias = syntheticExternsAtEnd.getFirstChild();
+    assertThat(injectedAlias).isNotNull();
```

Comment:
This assertion seems to be failing for me. Is it not for you?

Reply:
It wasn't failing - but is now due to the synthetic externs changing.